pkg/apis/core/helper/helpers.go

```diff
@@ -167,3 +169,61 @@ func FindWorkerByName(workers []core.Worker, name string) *core.Worker {
 	}
 	return nil
 }
+
+// GetRemovedVersions finds versions that have been removed in the old compared to the new version slice.
+// returns a map associating the index in the old version slice with the version and a string set containing the versions.
+func GetRemovedVersions(old []core.ExpirableVersion, new []core.ExpirableVersion) (map[int]core.ExpirableVersion, sets.String) {
+	return getVersionDiff(old, new)
+}
+
+// GetAddedVersions finds versions that have been added in the new compared to the new version slice.
+// returns a map associating the index in the new version slice with the version and a string set containing the versions.
+func GetAddedVersions(old []core.ExpirableVersion, new []core.ExpirableVersion) (map[int]core.ExpirableVersion, sets.String) {
+	return getVersionDiff(new, old)
+}
+
+// getVersionDiff gets versions that are in v1 but not in v2.
+// Returns versions mapped to its index in v1 and also a string set containing the versions
+func getVersionDiff(v1, v2 []core.ExpirableVersion) (map[int]core.ExpirableVersion, sets.String) {
+	v2Versions := make(map[string]struct{}, len(v2))
+	for _, x := range v2 {
+		v2Versions[x.Version] = struct{}{}
+	}
+	diff := make(map[int]core.ExpirableVersion, len(v1))
```

Comment:
You don't know at this point, how long the diff map will be.
```suggestion
	diff := map[int]core.ExpirableVersion{}
```